### DISCUSSION TITLE
web components for miniPaint?

### DISCUSSION THREAD

**Discussion by ghost** [viliusle/miniPaint#346] [Q&A] [answered]
Hi.

#### Idea
I would like to suggest the idea of web components for miniPaint in the ui/ux part.

#### what is web components?
Web Components is a suite of different technologies allowing you to create reusable custom elements — with their functionality encapsulated away from the rest of your code — and utilize them in your web apps.

#### why?
1. My idea is that the user chooses the resources that miniPaint will have.
2. The advantage is that you can have a lighter version in miniPaint.
3. Today there are things like web components, I think of using web components to compose each miniPaint interface in an isolated, independent way. This allows for greater security, auditability
4. this allows to have widgets, gadgets, plugins, themes in the ui/ux part.
5. this makes life easier for the developer and the maintenance of a open source project (large, small or intermediate).
6. miniPaint works online through the browser, there is a web components feature to make the ui/ux experience better

#### question/feedback
What do you all think of the idea?

**Comment by viliusle** [comment, marked as answer]
I am not sure that it would help. Web components are good for reusable units, but on miniPaint, lots of things is depends on framework itself like layers, zoom, shared libraries.

**Reply by ghost** [reply]
Hi viliusle 

> I am not sure that it would help. Web components are good for reusable units, but on miniPaint, lots of things is depends on framework itself like layers, zoom, shared libraries.

1. Things like web components work with IPFS - *That's an advantage I forgot to mention.*
2. As I said at the beginning web components would be for the ui/ux interface and not for the js code
